apps/builder/app/builder/shared/assets/use-assets.tsx

```diff
@@ -103,7 +103,22 @@ const assetContainersStore = computed(
         });
       }
     }
-    return assetContainers;
+
+    return assetContainers.sort((container1, container2) => {
+      if (container1.status === "uploading") {
+        return -1;
+      }
+      if (container2.status === "uploading") {
+        return 1;
+      }
+      if (container1.status === container2.status) {
```

Comment:
order of conditions matters. If you check uploading then createdAt check won't run and your sort first random uploading instead of newest one.

Reply:
we need a sorting utility that is simpler, this is a mess

Reply:
It's specific to the use case. Nothing to abstract really.